src/legacy/btchip_apdu_hash_input_finalize_full.c

```diff
@@ -70,12 +70,16 @@ static bool check_output_displayable() {
     isOpCall =
         btchip_output_script_is_op_call(btchip_context_D.currentOutput + 8,
           sizeof(btchip_context_D.currentOutput) - 8);
-    if (((G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
+    if ((((G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
          !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
          !isP2sh && !(nullAmount && isOpReturn) && !isOpCreate && !isOpCall) ||
         (!(G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
          !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
-         !isP2sh && !(nullAmount && isOpReturn))) {
+         !isP2sh && !(nullAmount && isOpReturn))) &&
+       (!(((G_coin_config->kind == COIN_KIND_PEERCOIN) &&
+         !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
+         !isP2sh && !(nullAmount && isOpReturn) &&
+         nullAmount)))) {
```

Comment:
The code was quite unreadable even before the PR, now I'm not able to verify it at all.
I suggest restructuring with the something similar to the following:
```C
  bool is_output_script_acceptable = false;
  if (G_coin_config->kind == COIN_KIND_FOO) {
    // TODO
  } else if (G_coin_config->kind == COIN_KIND_BAR) {
    // TODO
  } else {
    // TODO
  }

  if (!is_output_script_acceptable) {
        PRINTF("Error : Unrecognized output script");
        THROW(EXCEPTION);
  }
```

Reply:
good idea, thank you, i've refactored this statement like you have asked and it did make it much more readable